### PR TITLE
Note Possible Injection Vulnerabilities and Prevention Techniques

### PR DESCRIPTION
**Pull request by cfromknecht**
This PR adds basic security recommendations for handling of node aliases and payment descriptions. Both fields are permitted to contain arbitrary user input, making them particularly susceptible to exploits via injection attacks if the input is controlled by an adversary. These commits make inline notes of the potential for these attacks if they are not properly mitigated and provide inline references for best practices in preventing various types of injection attacks.

If anyone has more resources to include or other relevant attacks that were missed, I would welcome more suggestions :)

### CODE REVIEW

File: 07-routing-gossip.md [hunk 1]
```diff
@@ -299,6 +299,25 @@ to be ordered in ascending order, unknown ones can be safely ignored.
 Future fields beyond `addresses` can still be added, optionally with
 padding within `addresses` if they require certain alignment.
 
+### Security Considerations for Node Aliases
+
+The node alias field is user-defined and has no cryptographic validation apart
```
Comment (cdecker) [inline 07-routing-gossip.md]: I don't fully understand the need to specify the cryptographic validation. You mean there is no way for others to manipulate on the fly?

File: 07-routing-gossip.md [hunk 2]
```diff
@@ -299,6 +299,25 @@ to be ordered in ascending order, unknown ones can be safely ignored.
...
+from checking that it is covered in the announcement's signature. This makes it
+particularly vulnerable to injection attacks, both in the process of rendering
```
Comment (cdecker) [inline 07-routing-gossip.md]: I'd say it's the attack vector and not itself vulnerable :wink:

File: 07-routing-gossip.md [hunk 3]
```diff
@@ -299,6 +299,25 @@ to be ordered in ascending order, unknown ones can be safely ignored.
...
+HTML/Javascript contexts, or any other dynamically interpreted rendering
+frameworks.  Similarly, consider using prepared statements, input validation,
+and/or escaping to protect against injection vulnerabilities against persistence
+engines that support SQL or other dynamically interpreted querying languages. 
+
+* [Stored and Reflected XSS Prevention](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet)
+* [DOM-based XSS Prevention](https://www.owasp.org/index.php/DOM_based_XSS_Prevention_Cheat_Sheet)
+* [SQL Injection Prevention](https://www.owasp.org/index.php/SQL_Injection_Prevention_Cheat_Sheet)
+
+Don't be like [Little Bobby Tables](https://xkcd.com/327/).
```
Comment (cdecker) [inline 07-routing-gossip.md]: Don't be like the school of Little Bobby Drop Tables ^^

File: 11-payment-encoding.md [hunk 4]
```diff
@@ -222,6 +222,30 @@ The `r` field allows limited routing assistance: as specified it only
 allows minimum information to use private channels, but it could also
 assist in future partial-knowledge routing.
 
+### Security Considerations for Payment Descriptions
+
+The payment description field is user-defined and has no cryptographic
+validation apart from checking that it is covered in the payment request's
+signature. This makes it particularly vulnerable to injection attacks, both in
```
Comment (rustyrussell) [inline 11-payment-encoding.md]: Remove `and has no....` it works well.

**Comment by cfromknecht** [comment]
Thank you for the feedback @cdecker and @rustyrussell! I've reworded to state that the user-defined fields provide a potential attack vector, it reads much clearer now IMO. I also updated to include "the school of" in the Little Bobby Tables references. 

Rebased and squashed the fixups on top of latest master ⚡️

**Comment by rustyrussell** [comment]
Applied, as per decision at https://docs.google.com/document/d/1SSRsT0iyRv2XtdCBlRU6WttBO1POo7e9lQPsEiJacYY

**Pull request merged**